main/vstring.h

```diff
@@ -30,7 +30,7 @@
 #endif
 #ifdef VSTRING_PUTC_MACRO
 #define vStringPut(s,c) \
-	(void)(((s)->length + 1 == (s)->size ? vStringResize ((s), (s)->size * 2) : 0), \
+	(void)(((s)->length + 1 == (s)->size ? vStringResize ((s), (s)->size * 2) : (void) 0 ), \
 	((s)->buffer [(s)->length] = (c)), \
 	((c) == '\0' ? 0 : ((s)->buffer [++(s)->length] = '\0')))
 #endif
```

Comment:
@masatake Yes your suggestion is equivalent.

But if I ought to rewrite the code I would write something like

```cpp
+#define vStringPut(s,c)                            \
+   do {                                        \
+       if ((s)->length + 1 >= (s)->size)        \
+           vStringResize ((s), (s)->size * 2); \
+
+       (s)->buffer [(s)->length] = (c);          \
+      (s)->length++;                                     \
+       (s)->buffer [(s)->length] = '\0';          \
+   } while (0);
```

As that is much more pleasant to read and should execute at the same speed.